doc/specs/#396 - Settings command.md

```diff
@@ -0,0 +1,175 @@
+
+1. winget settings
+2. If settings file can be parsed copy to settings.backup
+3. Open settings file
+4. Wait for next command
+5. If settings file can be parsed use it otherwise use settings.backup and warn user.
+
+This mechanism allows to be resilient for mistakes done the settings file.
+
+### First time use
+
+Since the settings file doesn't exist at this time, we can't force the creation of it at WinGet install time. This means that WinGet must still work without the existence of the file. The file will be created when the user runs the settings command it it doesn't exists. The new file will reference the settings documentation. 
+
+### Settings
+
+Two settings will be implemented for the sake of the discussion and to not provide a setting feature without actual settings.
+
+### Progress Bar Visual Style
+
+This setting will specify the color of the progress bar that WinGet displays. It will only support the current visual style that WinGet already support and no new ones will be added for now.
+
+Possible values:
+1. accent (default)
+1. rainbow
+1. plain
+
+YAML
+```
+Visual:
+  - ProgressBar: accent
+```
+
+JSON
+```
+  "Visual": [
+    {
+      "ProgressBar": "accent"
+    }
+  ]
+```
+
+#### Override
+
+The user must be able to keep using the preferred visual style via arguments as it is now. This has more priority than the style defined in the settings file, because is saying it want that style for the current run.
+
+#### Special request
+
+The author of this document firmly believes that `--plain` must be replaced with `--retro`.
+
+### Source auto update
+
+Currently, WinGet updates the source after 5 minutes. This setting will will enable users to set the timeout in minutes.
+
+Value must be integers with a minimum of 0. An arbitrary limit limit can be set but is not strictly necessary. A value of 0 indicates no update.
+
+YAML
+```
+Source:
+  - AutoUpdateIntervalInMinutes: 5
+```
+
+JSON
+```
+  "Source": [
+    {
+      "AutoUpdateIntervalInMinutes": 5
+    }
+  ]
+```
+
+Having the unit defined in the property makes it self documented and avoids the pain of opening the settings file to see the unit.
+
+When the value is set to 0 it is the user responsibility to update the source if needed via `winget source update`
+
+## UI/UX Design
+
+Executing `winget settings` will open settings file in the default text editor the user has for the file extension.
+
+## Capabilities
+
+It will allow winget to set and define any new settings in the future.
+
+### Accessibility
+
+Any description or output text that is added by consequence of this feature will need to be localized. Moreover, this feature allows a mechanism to add accessibility settings in the future.
+
+### Security
+
+This should not introduce any _new_ security concerns.
+
+### Reliability
```

Comment:
Consider here, or somewhere, describing what will happen if the file can be parsed, but some value is outside of its defined range.

Will we ignore individual settings?  Or will we treat it as an error all up and load the backup?